cmd/backup-manager/app/restore/manager.go

```diff
@@ -295,8 +296,11 @@ func (rm *Manager) performRestore(ctx context.Context, restore *v1alpha1.Restore
 		TimeCompleted: &metav1.Time{Time: finish},
 		CommitTs:      &ts,
 	}
-	return rm.StatusUpdater.Update(restore, &v1alpha1.RestoreCondition{
-		Type:   v1alpha1.RestoreComplete,
-		Status: corev1.ConditionTrue,
-	}, updateStatus)
+	// try best effort to guarantee restore status is updated.
+	return retry.OnError(retry.DefaultRetry, func(e error) bool { return e != nil }, func() error {
```

Comment:
Is it better to define a function `rm.StatusUpdater.RetryUpdate()`?

Reply:
BTW, there are many codes that will update the status for the backup and restore CR (`grep -r "Update(" cmd`), maybe it's better to update the logic of the `Update` function.

Reply:
I'm trying to update the logic inside `Update` function at first, but I think it's not the client's responsibilty to retry on errors like conn timeout.

Do we need to make `Update` function to have the `guaranteeUpdate` semantics here?